packages/generator/test/parser/edmx-to-vdm-v4.spec.ts

```diff
@@ -0,0 +1,242 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { transformEntitiesV4 } from '../../src/parser/edmx-to-vdm-v4';
+import { ServiceNameFormatter } from '../../src/service-name-formatter';
+import {
+  EdmxProperty,
+  EdmxFunctionImportBase
+} from '../../src/parser/parser-types-common';
+import {
+  EdmxEntitySet,
+  EdmxEntityType,
+  EdmxMetadata
+} from '../../src/parser/parser-types-v4';
+import { transformComplexTypes } from '../../src/parser/edmx-to-vdm-common';
+import { ParsedServiceMetadata } from '../../src/parser/parsed-service-metadata';
+
+describe('edmx-to-vdm-v4', () => {
+  it('transforms collection type properties for primitive types', () => {
+    const service = createTestServiceData(
+      [
+        createEntityType('TestEntityType', [
+          ['CollectionProperty', 'Collection(Edm.String)', false]
+        ])
+      ],
+      [createTestEntitySet('TestEntity', 'TestEntityType')]
+    );
+
+    const entity = transformEntitiesV4(service, [], getFormatter(service))[0];
+    expect(entity.properties[0]).toMatchObject({
+      isCollection: true,
+      edmType: 'Edm.String',
+      jsType: 'string',
+      fieldType: 'StringField'
+    });
+  });
+
+  it('transforms collection type properties for complex types', () => {
+    const service = createTestServiceData(
+      [
+        createEntityType('TestEntityType', [
+          ['CollectionProperty', 'Collection(namespace.TestComplexType)', false]
+        ])
+      ],
+      [createTestEntitySet('TestEntity', 'TestEntityType')]
+    );
+
+    const formatter = getFormatter(service);
+    const vdmComplexTypes = transformComplexTypes(
+      service.edmx.complexTypes,
+      formatter,
+      new Set()
+    );
+
+    const entity = transformEntitiesV4(service, vdmComplexTypes, formatter)[0];
+    expect(entity.properties[0]).toMatchObject({
+      isCollection: true,
+      isComplex: true,
+      edmType: 'namespace.TestComplexType',
```

Comment:
Ah, I found the test. So namespace is part of the type?

Reply:
Yeah, here it refers to the type as found in the edmx file. I added some explicit tests.